code/game/objects/structures/inflatable.dm

```diff
@@ -0,0 +1,292 @@
+	anchored = TRUE
+	opacity = FALSE
+
+	icon = 'icons/obj/inflatable.dmi'
+	icon_state = "wall"
+
+	var/health = 50.0
+
+/obj/structure/inflatable/Initialize(location)
+	..()
+	air_update_turf(TRUE)
+
+/obj/structure/inflatable/Destroy()
+	air_update_turf(TRUE)
+	return ..()
+
+/obj/structure/inflatable/CanPass(atom/movable/mover, turf/target, height=0)
+	return FALSE
+
+/obj/structure/inflatable/CanAtmosPass(turf/T)
+	return !density
+
+/obj/structure/inflatable/bullet_act(var/obj/item/projectile/Proj)
+	health -= Proj.damage
+	..()
+	if(health <= 0)
+		deflate(TRUE)
+	return
+
+/obj/structure/inflatable/ex_act(severity)
+	switch(severity)
+		if(1.0)
+			qdel(src)
+			return
+		if(2.0)
+			deflate(TRUE)
+			return
+		if(3.0)
+			if(prob(50))
+				deflate(TRUE)
+				return
+
+/obj/structure/inflatable/blob_act()
+	deflate(TRUE)
+
+/obj/structure/inflatable/attack_hand(mob/user as mob)
+	add_fingerprint(user)
+	return
+
+/obj/structure/inflatable/proc/attack_inflatable(mob/user as mob, damage = 0)	//used by attack_alien, attack_animal, and attack_slime
+	health -= damage
+	if(health <= 0)
+		user.visible_message("<span class='danger'>[user] tears open [src]!</span>")
+		deflate(TRUE)
+	else	//for nicer text~
+		user.visible_message("<span class='danger'>[user] tears at [src]!</span>")
+
+/obj/structure/inflatable/attack_alien(mob/user as mob)
+	if(islarva(user))
+		return
+	attack_inflatable(user, 15)
+
+/obj/structure/inflatable/attack_animal(mob/user as mob)
+	if(!isanimal(user))
+		return
+	var/mob/living/simple_animal/M = user
+	if(M.melee_damage_upper <= 0)
+		return
+	attack_inflatable(M, M.melee_damage_upper)
+
+/obj/structure/inflatable/attack_slime(mob/user as mob)
+	var/mob/living/simple_animal/slime/S = user
+	if(!S.is_adult)
+		return
+	attack_inflatable(user, rand(10, 15))
+
+/obj/structure/inflatable/attackby(obj/item/W as obj, mob/user as mob, params)
+	if(!istype(W))
+		return
+	if(is_pointed(W))
+		visible_message("<span class='danger'>[user] pierces [src] with [W]!</span>")
+		deflate(TRUE)
+	if(W.damtype == BRUTE || W.damtype == BURN)
+		hit(W.force)
+		..()
+	return
+
+/obj/structure/inflatable/proc/hit(var/damage, var/sound_effect = TRUE)
+	health = max(0, health - damage)
+	if(sound_effect)
+		playsound(loc, 'sound/effects/Glasshit.ogg', 75, 1)
+	if(health <= 0)
+		deflate(TRUE)
+
+/obj/structure/inflatable/AltClick()
+	if(usr.stat || usr.restrained())
+		return
+	if(!Adjacent(usr))
+		return
+	deflate()
+
+/obj/structure/inflatable/proc/deflate(var/violent = TRUE)
+	playsound(loc, 'sound/machines/hiss.ogg', 75, 1)
+	if(violent)
+		visible_message("[src] rapidly deflates!")
+		var/obj/item/inflatable/torn/R = new /obj/item/inflatable/torn(loc)
+		src.transfer_fingerprints_to(R)
+		qdel(src)
+	else
+		visible_message("[src] slowly deflates.")
+		spawn(50)
+			var/obj/item/inflatable/R = new /obj/item/inflatable(loc)
+			src.transfer_fingerprints_to(R)
+			qdel(src)
+
+/obj/structure/inflatable/verb/hand_deflate()
```

Comment:
For optimal UX, I'd like it to be possible to deflate them with an Alt+click, with a complementary bit in the examine description.

Reply:
>:C

Reply:
I think there has been discussion / contention before about this, either on paradise or baystation. Making it an alt-click thing makes it a bit too easy IMO to deflate by accident and cause a pressure casualty, and consequently adds a fair bit of deniability as well for griefers.

I suppose I could lock it behind a prompt that follow the alt-click, but then... what's the point?

Reply:
I mean, alt-click is the shortcut for viewing the floor tile, something a lot of people use, at least I do. I know I would probably accidentally deflate it a bunch of times if I added an alt-click thing. Maybe alt-double click but I don't know how the game supports that and it might become snowflakey for this one feature.

Reply:
A prompt would be an improvement over using the right-click menu. A lot of work has been done to move towards a more consistent and useful interface and that thing is not it.